src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/optvector/SubtypeOnlyFloatVectorToArrayCodec.java

```diff
@@ -0,0 +1,180 @@
+    if (bytes == null || bytes.remaining() == 0) {
+      throw new IllegalArgumentException(
+          "Input ByteBuffer must not be null and must have non-zero remaining bytes");
+    }
+    // TODO: Do we want to treat this as an error?  We could also just ignore any extraneous bytes
+    // if they appear.
+    if (bytes.remaining() % ELEMENT_SIZE != 0) {
+      throw new IllegalArgumentException(
+          String.format("Input ByteBuffer should have a multiple of %d bytes", ELEMENT_SIZE));
+    }
+    ByteBuffer input = bytes.duplicate();
+    int elementCount = input.remaining() / 4;
+    float[] array = new float[elementCount];
+    for (int i = 0; i < elementCount; i++) {
+      deserializeElement(input, array, i, protocolVersion);
+    }
+    return array;
+  }
+
+  /**
+   * Write the {@code index}th element of {@code array} to {@code output}.
+   *
+   * @param output The ByteBuffer to write to.
+   * @param array The array to read from.
+   * @param index The element index.
+   * @param protocolVersion The protocol version to use.
+   */
+  protected void serializeElement(
+      @NonNull ByteBuffer output,
+      @NonNull float[] array,
+      int index,
+      @NonNull ProtocolVersion protocolVersion) {
+    output.putFloat(array[index]);
+  }
+
+  /**
+   * Read the {@code index}th element of {@code array} from {@code input}.
+   *
+   * @param input The ByteBuffer to read from.
+   * @param array The array to write to.
+   * @param index The element index.
+   * @param protocolVersion The protocol version to use.
+   */
+  protected void deserializeElement(
+      @NonNull ByteBuffer input,
+      @NonNull float[] array,
+      int index,
+      @NonNull ProtocolVersion protocolVersion) {
+    array[index] = input.getFloat();
+  }
+
+  @NonNull
+  @Override
+  public String format(@Nullable float[] value) {
+    return value == null ? "NULL" : Arrays.toString(value);
+  }
+
+  @Nullable
+  @Override
+  public float[] parse(@Nullable String str) {
+    /* TODO: Logic below requires a double traversal through the input String but there's no other obvious way to
+     * get the size.  It's still probably worth the initial pass through in order to avoid having to deal with
+     * resizing ops.  Fortunately we're only dealing with the format/parse pair here so this shouldn't impact
+     * general performance much. */
+    if ((str == null) || str.isEmpty()) {
+      throw new IllegalArgumentException("Cannot create float array from null or empty string");
+    }
+    Iterable<String> strIterable =
+        Splitter.on(", ").trimResults().split(str.substring(1, str.length() - 1));
+    float[] rv = new float[Iterators.size(strIterable.iterator())];
+    Iterator<String> strIterator = strIterable.iterator();
+    for (int i = 0; i < rv.length; ++i) {
+      String strVal = strIterator.next();
+      // TODO: String.isBlank() should be included here but it's only available with Java11+
```

Comment:
Since we are using Java 21, why don't we use String.isBlank() here?

Reply:
Code was first written for Java CQL driver (by Brett). I can change that, thanks.